### PR TITLE
Fix sandboxed process tests

### PR DESCRIPTION
**Pull request by almiliMSFT**
[AB#1519309](https://dev.azure.com/mseng/708e929f-6bd5-415a-8daf-25b1dac08dd8/_workitems/edit/1519309)
[AB#1519677](https://dev.azure.com/mseng/708e929f-6bd5-415a-8daf-25b1dac08dd8/_workitems/edit/1519677)

### CODE REVIEW

File: Public/Src/Engine/Processes/SandboxedProcess.MacKext.cs [hunk 1]
```diff
@@ -124,22 +124,34 @@ public SandboxedProcessMacKext(SandboxedProcessInfo info, bool ignoreReportedAcc
...
-        private async Task StartAsync()
+        protected override System.Diagnostics.Process CreateProcess()
         {
-            base.CreateAndSetUpProcess();
+            var process = base.CreateProcess();
 
-            Process.StartInfo.FileName = "/bin/sh";
-            Process.StartInfo.Arguments = string.Empty;
-            Process.StartInfo.RedirectStandardInput = true;
-            Process.Start();
```
Comment (almiliMSFT) [inline Public/Src/Engine/Processes/SandboxedProcess.MacKext.cs]: The main problem was that this class used to start the process (call `Process.Start()`) while the base class used `m_processExecutor` for other things (like wait for the process to exit, compute process start/exit time, etc.).  Now all that process lifetime management goes through the `AsyncProcessExecutor`

File: Public/Src/Engine/UnitTests/Processes/SandboxedProcessTestBase.cs [hunk 2]
```diff
@@ -35,7 +37,15 @@ public SandboxedProcessTestBase(ITestOutputHelper output) : base(output)
             var envVars = Override(
                 BuildParameters.GetFactory().PopulateFromEnvironment().ToDictionary(),
                 overrideEnvVars);
-            
+
+            var testMethodFrame = new System.Diagnostics.StackTrace()
```
Comment (dannyvv) [inline Public/Src/Engine/UnitTests/Processes/SandboxedProcessTestBase.cs]: >testMethodFrame [](start = 16, length = 15)

Maybe this should go into the BuildXLTest baseclass?

Comment (almiliMSFT) [reply Public/Src/Engine/UnitTests/Processes/SandboxedProcessTestBase.cs]: I'll check and move it there if possible

Comment (almiliMSFT) [reply Public/Src/Engine/UnitTests/Processes/SandboxedProcessTestBase.cs]: extracted `DiscoverCurrentlyExecutingXunitTestMethodFQN()` method and used it in other places where `SandboxedProcessInfo` is created

**Pull request merged**